python/simulation.py

```diff
@@ -3259,23 +3256,34 @@ def get_array_metadata(self, vol=None, center=None, size=None, dft_cell=None,
         $$ \\int_{\\mathcal V} Q(\\mathbf{x})d\\mathbf{x} \\approx \\sum_{n} w_n Q_n.$$
 
         This is a 1-, 2-, or 3-dimensional integral depending on the number of dimensions
-        in which $\\mathcal{V}$ has zero extent. If the $\\{Q_n\\}$ samples are stored in an
+        in which $\\mathcal{V}$ has zero extent. If the $Q_n$ samples are stored in an
         array `Q` of the same dimensions as `w`, then evaluating the sum on the RHS is
         just one line: `np.sum(w*Q).`
 
         A convenience parameter `dft_cell` is provided as an alternative to `vol` or
-        `center/size`; set `dft_cell` to a `dft_flux` or `dft_fields` object to define the
-        region covered by the array. If the `dft` argument is provided then all other
-        arguments (`vol`, `center`, and `size`) are ignored. If no arguments are provided,
-        then the entire cell is used.
+        `center`/`size`. Set `dft_cell` to a `dft_flux` or `dft_fields` object to define the
+        region covered by the array. If the `dft_cell` argument is provided then all other
+        arguments related to the spatial region (`vol`, `center`, and `size`) are ignored.
+        If no arguments are provided, then the entire cell is used.
+
+        By default, for empty dimensions of the grid slice `get_array_metadata` will return
+        *two* elements corresponding to the nearest Yee grid points. Setting `collapse=True`
+        will collapse the empty dimensions into a *single* element using a linear interpolation
+        of the nearest Yee grid points. If the empty dimension of the grid slice exactly
+        overlaps the Yee grid points, the array slice that is returned will *always* have
+        a single element independent of the value for `collapse`.
+
+        If `return_pw=True`, the return value is a 2-tuple `(p,w)` where `p` (points) is a
+        list of `mp.Vector3`s with the same dimensions as `w` (weights). Otherwise, by
+        default the return value is a 4-tuple `(x,y,z,w)`.
```

Comment:
Just a drive by comment here... it looks like the return type is actually a list, not a tuple. This could be confusing for someone trying to unpack the return values.

Reply:
Right, I think we should return a tuple here, just to follow the usual Python convention for multiple return values.  See the 1-line change below.